bin/ddg2pca

```diff
@@ -0,0 +1,405 @@
+
+    # Extracting the group for each label and putting it in a list
+    residue_grouping = [list(pos_dic[i].keys())[0] for i in range(len(res_id_labels))]
+
+    # Converting the list to a numpy array
+    residue_grouping = np.array(residue_grouping)
+
+    ### Group processing
+    options_group = options.group_selection #change name of residue nature
+
+    # Filter res_ids_str and labels using indices.
+    filtered_res_ids_str = filter_by_group(res_ids_str, pos_dic, options_group)
+    filtered_labels = filter_by_group(labels, pos_dic, options_group)
+    filtered_residue_grouping = filter_by_group(residue_grouping, pos_dic, options_group)
+
+    # Determine which labels were removed.
+    removed_labels = set(labels) - set(filtered_labels)
+
+    # Remove the same labels from textlist.
+    filtered_textlist = [label for label in textlist if label[0] not in removed_labels]
+   
+    data = []
+
+    log.info("The following files will be considered: %s" % ", ".join(fnames))
+    for i,f in enumerate(fnames):
+        try:
+            # Convert the data to float as it's loaded
+            data.append(np.array(parse_ddg_file("%s%s" % (options.ddg_dir, f), reslist=res_order, )).astype(float))
+        except (IOError, TypeError, ValueError):
+            log.error("Failed to read input file %s%s" % (options.ddg_dir, f))
+            exit(1)
+
+    
+    csv_table = np.asarray(["WT residue type", "chain ID", "Residue #"])
+
+    for i,r in enumerate(res_order):
+        csv_table = np.append(csv_table, r)
+        
+            
+    for i, resset in enumerate(filtered_res_ids_str):
+        chain = ["".join([res[1] for res in resset])]
+
+        lines = np.append(resset[0][2:], data[i])
+        lines = np.append(chain, lines)
+        lines = np.append(resset[0][0], lines)
+        csv_table = np.vstack((csv_table, lines))
+
+    
+    column_headers = csv_table[0]
+    column_headers = np.concatenate((np.array(["Residue_label"]), column_headers[3:]))
+    data = csv_table[1:]
+
+
+    # Combining the first three columns into one (ojo)
+    combined_column = np.core.defchararray.add(np.core.defchararray.add(data[:,0], data[:,1]), data[:,2])
+    combined_column = combined_column.reshape(-1, 1)  # reshape to 2D column vector
+
+    # Now delete the first three columns
+    data = np.delete(data, [0,1,2], axis=1)
+
+    # Concatenate the combined_column with the modified array
+    data = np.hstack((combined_column, data))
+
+    # Replacing the res_str in the data by the corresponding labels 
+    for i, label in enumerate(filtered_labels):
+        data[i, 0] = label
+
+    return data, column_headers, filtered_labels, filtered_textlist, filtered_residue_grouping, color_dict
+
+    # np.savetxt(f"{options.name_outfile}.csv", csv_table, delimiter=",", fmt='%s')  
+def perform_pca(data):
+    
+    data_scaled = (data[:, 1:].astype(float) - np.mean(data[:, 1:].astype(float), axis=0)) / np.std(data[:, 1:].astype(float), axis=0)
```

Comment:
it would be good to have an option not to rescale data (while I think we should always center them). After all, all our data sets are on the same scale by definition